src/web/lib/braze/LocalMessageCache.ts

```diff
@@ -0,0 +1,76 @@
+import type appboy from '@braze/web-sdk-core';
+import { SlotName } from './BrazeMessages';
+
+const localStorageKeyBase = 'gu.brazeMessageCache';
+
+type Message = appboy.InAppMessage;
+
+interface Store {
+	getItem: (key: string) => string | null;
+	setItem: (key: string, value: string) => void;
+	removeItem: (key: string) => void;
+}
+
+const keyFromSlotName = (slotName: SlotName): string =>
+	`${localStorageKeyBase}.${slotName}`;
+
+class LocalMessageCache {
```

Comment:
I wonder if this class is missing a way to clear out expired messages? We'll want to consider the time a message should be held on for, but unlike cookies LocalStorage does not have an in-built expiry mechanism.

Should we make sure that the message's date is part of the stored information, and include a method for deleting expired messages? That method doesn't need to necessarily be "public", it can be an internal implementation detail to keep the queue containing only relevant messages. In particular, call to `peek` and `shift` should never return expired messages so that client code can safely use the cache without the idea of expiry leaking out into the rest of the codebase.

Reply:
Yep I'm on the same page with all this, just didn't want to cram too much into this first PR.

Reply:
Having thought about this a little more, the expiration is such a fundamental part of the cache that I don't think it makes sense to consider it without it, even as a first step. So I'll work on adding that functionality to this PR.

Reply:
I started added expiry this afternoon but haven't finished the tests yet - I expect some checks may fail in the mean time.

Reply:
This is now done.